ocaml/typing/typetexp.ml

```diff
@@ -465,7 +465,17 @@ and transl_type_aux env policy mode styp =
         match args with
         | (l, arg_mode, arg) :: rest ->
           check_arg_type arg;
-          let arg_cty = transl_type env policy arg_mode arg in
+          let arg_cty = (
+            match arg.ptyp_desc with
+            | Ptyp_extension ({txt="src_pos"; _}, _payload) ->
+              (* CR src_pos: This won't work correctly in Untypeast *)
+              let path = Predef.path_lexing_position in
+              let lid = Longident.Lident "lexing_position" in
+              let constr = newconstr path [] in
+              ctyp (Ttyp_constr (path, {txt=lid; loc=Location.none}, [])) constr
```

Comment:
Creating this bit of typedtree should basically be unnecessary, because the information is redundant with the information in the `arg_label`. So, to avoid all this, I see two ways forward:

1. Create a new constructor of `Typedtree.core_type_desc` to represent `[%src_pos]`.
2. Edit `arg_label` to also carry a type in all the cases that aren't `Position`. Then `Ttyp_arrow` would just store an `arg_label` and a `core_type`. And so this arg type would never need to be computed.

Option 2 is somehow more elegant, but option 1 is probably more expedient. Let's go with option 1 for now, maybe returning to option 2 later.

Reply:
Discussed in call, option 2 will be implemented in separate PR

Reply:
Changed our minds and going with option 1